src/server/pfs/server/local_block_api_server.go

```diff
@@ -215,23 +215,32 @@ func (s *localBlockAPIServer) readDiff(diff *pfsclient.Diff) (*pfsclient.DiffInf
 	return result, nil
 }
 
-func scanBlock(scanner *bufio.Scanner) (*pfsclient.BlockRef, []byte, error) {
+func readBlock(reader *bufio.Reader) (*pfsclient.BlockRef, []byte, error) {
 	var buffer bytes.Buffer
 	var bytesWritten int
 	hash := newHash()
-	for scanner.Scan() {
-		// they take out the newline, put it back
-		bytes := append(scanner.Bytes(), '\n')
+	for {
+		// We don't use ReadBytes or ReadString because they allocate new buffers
+		// whereas ReadLine only uses an internal buffer
+		bytes, isPrefix, err := reader.ReadLine()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, nil, err
+		}
+		// Append a newline if we have reached the end of the line
+		if !isPrefix {
+			bytes = append(bytes, '\n')
+		}
 		buffer.Write(bytes)
 		hash.Write(bytes)
 		bytesWritten += len(bytes)
+		// len(bytes) == 0 when we've
 		if bytesWritten > blockSize {
 			break
 		}
```

Comment:
It seems like this could lead to a long line spanning blocks, what if we check that `bytesWritten > blockSize && !isPrefix`?